src/ZPublisher/WSGIPublisher.py

```diff
@@ -283,10 +283,16 @@ def publish_module(environ, start_response,
 
     path_info = environ.get('PATH_INFO')
     if path_info and PY3:
-        # The WSGI server automatically treats the PATH_INFO as latin-1 encoded
-        # bytestrings. Typically this is a false assumption as the browser
-        # delivers utf-8 encoded PATH_INFO. We, therefore, need to encode it
-        # again with latin-1 to get a utf-8 encoded bytestring.
+        # BIG Comment, see discussion at
+        # https://github.com/zopefoundation/Zope/issues/575
+        #
+        # The WSGI server automatically treats headers, including the
+        # PATH_INFO, as latin-1 encoded bytestrings, according to PEP-3333.
+        # Typically this is a false assumption nowadays as the browser delivers
+        # utf-8 url-encoded PATH_INFO, like '/t%C3%A4st', resulting in '/tÃ¤st'
+        # after latin-1 decode by the wsgi-server at this point. We, therefore,
+        # need to encode it again with latin-1 to get a utf-8 encoded
```

Comment:
This sentence makes no sense. Encoding with latin-1 gives you a latin-1 encoded byte string.

Reply:
not if the original byte string was something else. `b'/t\xc3\xa4st'.decode('latin-1').encode('latin-1')` is still a utf-8 encoded byte string.

Reply:
That claim is only true for those unicode code points shared by latin-1 and utf-8. If there is any mentioning in PEP-3333 that you will only ever get those code points in the WSGI pipeline please mention that. Otherwise the general claim is wrong and confusing (to me at least).